index.js

```diff
@@ -70,7 +70,10 @@ class SHACLValidator {
   }
 
   // Exposed to be available from validation functions as `SHACL.nodeConformsToShape`
-  nodeConformsToShape(focusNode, shapeNode, engine = this.validationEngine.clone()) {
+  nodeConformsToShape(focusNode, shapeNode, engine = this.validationEngine.clone(), resetCheckedNodes = false) {
```

Comment:
Since the engine is cloned, maybe you'd move the `checkedNodes` set there and create a new instance when cloning?